### ISSUE TITLE
Don't redefine Release

### ISSUE THREAD

**Issue by kpu** [marian-nmt/marian-dev#868]
Unfortunately Marian redefines Release to include `-g`.  

cmake has Release and RelWithDebInfo:
Release does not have debugging info.  
RelWithDebInfo has debugging info.  
If I wanted debug symbols, I can compile with `-DCMAKE_BUILD_TYPE=RelWithDebInfo`.  

This breaks projects that want to use Marian as a submodule.  If I compile with Release, `-g` is turned on and I get large binaries.  You might say use Slim but Slim isn't defined outside Marian so I get an unoptimized version of the parent repository.  Or I have to copy Marian's non-standard definitions to the parent repository.

**Comment by emjotde** [comment]
hm, would turning that off make the stack traces unusable? Very conflicted on that one.

**Comment by kpu** [comment]
Would you accept making the default build type `RelWithDebInfo` while returning `Release` to its former glory?

**Comment by emjotde** [comment]
Yeah, was thinking in that direction. I need to familiarize myself with that and also coordinate with people internally who are using Marian as a submodule right now.
